### ISSUE TITLE
Reference docs?

### ISSUE THREAD

**Issue by agurtovoy** [typeorm/typeorm#2440]
**Issue type:**

[x] question
[ ] bug report
[ ] feature request
[x] documentation issue

**Database system/driver:**

[ ] `cordova`
[ ] `mongodb`
[ ] `mssql`
[ ] `mysql` / `mariadb`
[ ] `oracle`
[ ] `postgres`
[ ] `sqlite`
[ ] `sqljs`
[ ] `react-native`

**TypeORM version:**

[x] `latest`
[ ] `@next`
[ ] `0.x.x` (or put your version here)

**Steps to reproduce or a small repository showing the problem:**

First of all, thanks for the great lib! I have a question on the docs.

I found that the currents docs are great at giving one an overall intro, but are somewhat lacking when it comes to the details of specific methods/options/etc. We find ourselves digging into the source code to answer questions about the library's interface more often than we'd like. The source code is nicely documented, but it's still a suboptimal experience.

Are there reference docs somewhere that I missed (besides the Decorators reference)?

**Comment by pleerock** [comment]
no, there is no. if you are talking about source code documentation then there is a tool called `typedoc` but I don't like it and TypeScript totally deserves something better. Im waiting for something better to come in TypeScript ecosystem.

Regarding to lack of docs on our site - then please feel free to contribute. Site documentation is definitely must be improved and we are welcomed to improvements!

**Comment by agurtovoy** [comment]
@pleerock Thanks for the quick response! Typedoc is one of the options I was wondering about. I tried it out last night after posting this issue, and while the output is far from perfect, I personally found the resulting docs quite useful in answering the API questions I had more quickly than I could before. 

Obviously, just one person's opinion, so please take it for what it's worth. For the time being, I posted the auto-generated docs online at http://typeorm.delightful.studio/

**Comment by vlapo** [comment]
@pleerock Why do you not like `typedoc`? It is about look&feelo or is there something more?

**Comment by vlapo** [comment]
I agree it is ugly. I'll check the possibilities of theme customization and try generate something usable.

**Comment by vlapo** [comment]
@pleerock I found compodoc today. It is used to angular projects but typescript project looks good too. It is also support md files but customization is little bit primitive but it is possible overwrite template files and change it to our purpose.

 - [main page](https://compodoc.app/)
 - [demo page](https://compodoc.github.io/compodoc-demo-todomvc-angular/index.html)
 - [available themes](https://compodoc.app/guides/themes.html)

What do you think about it? Maybe only API reference doc, but I think after small update it is suitable to main page too.

**Comment by vlapo** [comment]
typedoc looks better ? 😄

**Comment by Rfrixy** [comment]
Been a while, any update on this? @pleerock

**Comment by 1valdis** [comment]
Up. I find it much easier to work with api references, rather than trying to find something in guides or type definitions.

**Comment by faizanqubaz** [comment]
sas
